src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -2821,17 +2821,18 @@ bool AdapterHandlerLibrary::generate_adapter_code(AdapterBlob*& adapter_blob,
 
   // Get a description of the compiled java calling convention and the largest used (VMReg) stack slot usage
   int comp_args_on_stack = SharedRuntime::java_calling_convention(sig_bt, regs, total_args_passed);
+  address entry_address[AdapterBlob::ENTRY_COUNT];
+  assert(AdapterBlob::ENTRY_COUNT == 4, "sanity");
```

Comment:
Why you need this assert? Do you use `4` instead of `ENTRY_COUNT` somewhere?

Reply:
AdapterBlob constructor was using numeric values to access values in the entry_offset. I update it to use enums.